bakery/checker_test.go

```diff
@@ -0,0 +1,933 @@
+	client3 := newClient(locator)
+	client3.addMacaroon(ts, "1.bob", client1.macaroons[ts]["authn"])
+	client3.addMacaroon(ts, "2.alice", client2.macaroons[ts]["authn"])
+
+	// We should authenticate as bob (because macaroons are presented ordered
+	// by "cookie" name)
+	authInfo, err = client3.do(BC, ts, bakery.LoginOp)
+	c.Assert(err, gc.IsNil)
+	c.Assert(authInfo.Identity, gc.Equals, simpleIdentity("bob"))
+	c.Assert(authInfo.Macaroons, gc.HasLen, 1)
+
+	// Try them the other way around and we should authenticate as alice.
+	client3 = newClient(locator)
+	client3.addMacaroon(ts, "1.alice", client2.macaroons[ts]["authn"])
+	client3.addMacaroon(ts, "2.bob", client1.macaroons[ts]["authn"])
+
+	authInfo, err = client3.do(BC, ts, bakery.LoginOp)
+	c.Assert(err, gc.IsNil)
+	c.Assert(authInfo.Identity, gc.Equals, simpleIdentity("alice"))
+	c.Assert(authInfo.Macaroons, gc.HasLen, 1)
+}
+
+func addFirstPartyCaveats(m *macaroon.Macaroon, ns *checkers.Namespace, caveats ...checkers.Caveat) {
+	for _, cav := range caveats {
+		m.AddFirstPartyCaveat(ns.ResolveCaveat(cav).Condition)
+	}
+}
+
+// resolveCaveats resolves all the given caveats with the
+// given namespace and includes the condition
+// from each one. It will panic if it finds a third party caveat.
+func resolveCaveats(ns *checkers.Namespace, caveats []checkers.Caveat) []string {
+	conds := make([]string, len(caveats))
+	for i, cav := range caveats {
+		if cav.Location != "" {
+			panic("found unexpected third party caveat")
+		}
+		conds[i] = ns.ResolveCaveat(cav).Condition
+	}
+	return conds
+}
+
+func macaroonConditions(caveats []macaroon.Caveat, allowThird bool) []string {
+	conds := make([]string, len(caveats))
+	for i, cav := range caveats {
+		if cav.Location != "" {
+			if !allowThird {
+				panic("found unexpected third party caveat")
+			}
+			continue
+		}
+		conds[i] = string(cav.Id)
+	}
+	return conds
+}
+
+func firstPartyMacaroonCaveats(conds ...string) []macaroon.Caveat {
+	caveats := make([]macaroon.Caveat, len(conds))
+	for i, cond := range conds {
+		caveats[i].Id = []byte(cond)
+	}
+	return caveats
+}
+
+func readOp(entity string) bakery.Op {
+	return bakery.Op{
+		Entity: entity,
+		Action: "read",
+	}
+}
+
+func writeOp(entity string) bakery.Op {
+	return bakery.Op{
+		Entity: entity,
+		Action: "write",
+	}
+}
+
+// opAuthorizer implements bakery.Authorizer by looking the operation
+// up in the given map. If the username is in the associated slice
+// or the slice contains "everyone", authorization is granted.
+type opAuthorizer map[bakery.Op][]string
+
+func (auth opAuthorizer) Authorize(ctxt context.Context, id bakery.Identity, ops []bakery.Op) (allowed []bool, caveats []checkers.Caveat, err error) {
+	return authorizerFunc(auth.authorize).Authorize(ctxt, id, ops)
+}
+
+func (auth opAuthorizer) authorize(id string, op bakery.Op) (bool, []checkers.Caveat, error) {
+	for _, allowedId := range auth[op] {
+		if allowedId == "everyone" || id == allowedId {
```

Comment:
You've got two constants you can use for everyone, and you still put in the string!